src/components/carousel/carousel.component.tsx

```diff
@@ -38,16 +54,18 @@ const BaseCarousel = ({
   onSlideChange,
   slideIndex,
   ...props
-}) => {
+}: CarouselProps) => {
   const [selectedSlideIndex, setSelectedSlideIndex] = useState(
     Number(slideIndex) || Number(initialSlideIndex)
   );
   const transitionDirection = useRef(NEXT);
-  const lastSlideIndexProp = useRef(props.slideIndex);
+  const lastSlideIndexProp = useRef(slideIndex);
   const id = guid();
```

Comment:
```suggestion
  const id = useMemo(() => guid());
```
It might be worth memorising this value so the id isn't recalculated on the next render